SPHINXsys/src/shared/particle_dynamics/solid_dynamics/solid_dynamics.cpp

```diff
@@ -676,6 +676,51 @@ namespace SPH
 			vel_n_[index_i] += dvel_dt_[index_i] * dt;
 		}
 		//=================================================================================================//
+		KirchhoffStressRelaxationFirstHalf::
+			KirchhoffStressRelaxationFirstHalf(BaseInnerBodyRelation* body_inner_relation)
+			: StressRelaxationFirstHalf(body_inner_relation),
+			J_to_minus_2_over_diemsnion_(*particles_->createAVariable<indexScalar, Real>("DeterminantTerm")),
+			stress_on_particle_(*particles_->createAVariable<indexMatrix, Matd>("StressOnParticle")),
+			inverse_F_T_(*particles_->createAVariable<indexMatrix, Matd>("InverseTransposedDeformation")) {};
+		//=================================================================================================//
+		void KirchhoffStressRelaxationFirstHalf::Initialization(size_t index_i, Real dt)
```

Comment:
suggest to write unit test on this function

Reply:
OK.